backend/inspirehep/records/marshmallow/literature/ui.py

```diff
@@ -185,18 +184,24 @@ def get_number_of_references(self, data):
         references = data.get("references")
         return self.get_len_or_missing(references)
 
-    def get_linked_book(self, data):
-        parent = get_parent_record(data)
-        if parent and "titles" in parent and "control_number" in parent:
-            endpoint = PidStoreBase.get_endpoint_from_pid_type(
-                PidStoreBase.get_pid_type_from_schema(data["$schema"])
-            )
-            endpoint_item = f"invenio_records_rest.{endpoint}_item"
-            ref = get_value(parent, "self.$ref") or url_for(
-                endpoint_item, pid_value=parent["control_number"], _external=True
-            )
-            return {**parent["titles"][0], "record": {"$ref": ref}}
-        return None
+
+    def get_linked_books(self, data):
+        parents = get_parent_records(data)
+        pages = get_pages(data)
+
+        linked_books_without_pages = [
+            {**parent["titles"][0], "record": {"$ref": self._get_parent_reference(parent, data)}}
+            for parent in parents
+            if parent and "titles" in parent and "control_number" in parent
+        ]
+
+        linked_books = merge_values(pages["page_start"], pages["page_end"], linked_books_without_pages)
```

Comment:
we don't need the `merge_values`, something like the following should do the same:


```python
merged_list = [
    {'start_page': start, 'end_page': end, 'record': record}
    for start, end, record in zip(pages_start, pages_end, records)
]

```